packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-rule.ts

```diff
@@ -169,7 +390,7 @@ export class ApplicationListenerRule extends cdk.Construct {
    */
   public readonly listenerRuleArn: string;
 
-  private readonly conditions: {[key: string]: string[] | undefined} = {};
+  private readonly conditions: {[key: string]: IListenerRuleCondition} = {};
```

Comment:
I think it's probably okay (and better), to treat the classes of conditions set using the `Condition` class and those set using key/value to be different. I'm not sure the overwrite behavior was ever a feature, more an implementation detail that was surfaced.

If you think the overwriting is a feature, then I guess feel free to retain it.

Otherwise, I'd have `conditions: Condition[]` and just append to it.

And then maybe there is also a `legacyConditions: Record<string, Condition>` for the conditions set using key and value, where we can have the overwrite behavior specifically for those situations where we had that as legacy behavior.

Reply:
> I'm not sure the overwrite behavior was ever a feature, more an implementation detail that was surfaced.

oh... I see.

I'll consider the design.